eth_abi/decoding.py

```diff
@@ -144,6 +145,7 @@ def decode(self, stream):
         start_pos = decode_uint_256(stream)
 
         stream.push_frame(start_pos)
+        assert self.tail_decoder is not None, "`tail_decoder is None"
```

Comment:
It's easy to ignore `assert`s in runtime code, so we tend to prefer to keep them only in tests. More info here: https://github.com/ethereum/snake-charmers-tactical-manual/blob/main/antipatterns.md#assertions-in-runtime-code

Reply:
update to use `AssertionError`